src/lang.ts

```diff
@@ -213,3 +214,20 @@ export function partial(targetFunction: (...args: any[]) => any, ...suppliedArgs
 		return targetFunction.apply(this, args);
 	};
 }
+
+export function createHandle(destructor: () => void): Handle {
+	return {
+		destroy: function () {
+			this.destroy = function () {};
+			destructor.call(this);
+		}
+	};
+}
+
+export function createCompositeHandle(...handles: Handle[]): Handle {
+	return createHandle(function () {
+		for (var i = 0, handle: Handle; (handle = handles[i]); ++i) {
```

Comment:
I did not think `for...of` was supported by IE. Should we go with use `Array.prototype.forEach` for the same code cleanliness instead?